packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts

```diff
@@ -119,7 +131,7 @@ export class StepFunctionsStartExecution extends sfn.TaskStateBase {
     const policyStatements = [
       new iam.PolicyStatement({
         actions: ['states:StartExecution'],
-        resources: [this.props.stateMachine.stateMachineArn],
+        resources: [this.stateMachineArn],
```

Comment:
StartExecution only has resource-level permissions on the unqualified state machine ARN. If `this.stateMachineArn` is a version or alias ARN, we need to strip the qualifier `:<versionId>` or `<:aliasName>` at the end